### PR TITLE
Eng 416 component usage improvement

### PR DESCRIPTION
**Pull request by firegloves**
Can anyone check my implementation please?

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: engine/src/main/java/org/entando/entando/web/page/model/PageSearchRequest.java [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 import org.entando.entando.web.common.model.RestListRequest;
 
+// TODO firegloves => can we change its name in something like CodeEntitySearchRequest in order to generify?
```
Comment (Kerruba) [inline engine/src/main/java/org/entando/entando/web/page/model/PageSearchRequest.java:18]: Let's keep it as it is for now

File: engine/src/test/java/org/entando/entando/aps/system/services/assertionhelper/GuiFragmentAssertionHelper.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+
+public class GuiFragmentAssertionHelper {
+//
+//
+//    assertEquals(WidgetMockHelper.WIDGET_1_CODE, ((GuiFragmentDtoSmall) fragmentUsage.get("widget")).getCode("code"));
```
Comment (Kerruba) [inline engine/src/test/java/org/entando/entando/aps/system/services/assertionhelper/GuiFragmentAssertionHelper.java:20]: Remove commented code

File: engine/src/test/java/org/entando/entando/aps/system/services/assertionhelper/PageAssertionHelper.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+public class PageAssertionHelper {
+
+
+    /**
+     * does assertions on the received PagedMetadata basing on the default PageMockHelper mocked data
+     * @param usageDetails
+     */
+    public static void assertUsageDetails(PagedMetadata<ComponentUsageEntity> usageDetails) {
+
+       assertUsageDetails(usageDetails, PageMockHelper.UTILIZERS, PageMockHelper.UTILIZERS.length, 1);
```
Comment (Kerruba) [inline engine/src/test/java/org/entando/entando/aps/system/services/assertionhelper/PageAssertionHelper.java:23]: Create a test for the published page scenario

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by firegloves** [comment]
Retest this please

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
All is well

File: engine/javacore.20200518.103907.32073.0001.txt [hunk 4]
```diff
@@ -0,0 +1,2256 @@
+0SECTION       TITLE subcomponent dump routine
```
Comment (Kerruba) [inline engine/javacore.20200518.103907.32073.0001.txt:1]: Remove this file and add a gitignore

File: engine/src/test/java/org/entando/entando/aps/system/services/mockhelper/PageMockHelper.java [hunk 5]
```diff
@@ -29,21 +29,44 @@
...
         page.setParentCode(PARENT_CODE);
         page.setGroup(GROUP);
         page.setModel(mockServicePageModel());
 
+        setWidgets(page, widgetCodes);
+
+        return page;
+    }
+
+//    public static Page mockTestPageMission(String... widgetCodes) {
```
Comment (Kerruba) [inline engine/src/test/java/org/entando/entando/aps/system/services/mockhelper/PageMockHelper.java:44]: Required or deletable?

**Comment by entando-service** [comment]
All is well

**Pull request merged**